OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -144,46 +153,53 @@ public void onClick(View v) {
             }
         });
 
-        setupRegistrationForm();
+        callServerToGetRegistrationForm();
+
         hideSoftKeypad();
         tryToSetUIInteraction(true);
     }
 
-    public void showAgreement(RegistrationAgreement agreement) {
-        boolean isInAppEULALink = false;
-        try {
-            Uri uri = Uri.parse(agreement.getLink());
-            if (uri.getScheme().equals("edxapp")
-                    && uri.getHost().equals("show_eula")) {
-                isInAppEULALink = true;
-            }
-        } catch (Exception ex) {
-            logger.error(ex);
+    public void callServerToGetRegistrationForm() {
+        if (!NetworkUtil.isConnected(this)) {
+            showAlertDialog(getString(R.string.no_connectivity),
+                    getString(R.string.network_not_connected));
+            return;
         }
 
-        if (isInAppEULALink) {
-            // show EULA license that is shipped with app
-            environment.getRouter().showWebViewActivity(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
-        } else {
-            // for any other link, open agreement link in a webview container
-            environment.getRouter().showWebViewActivity(this, agreement.getLink(), agreement.getText());
-        }
+        final GetRegistrationFormTask task = new GetRegistrationFormTask(this){
+            @Override
+            protected void onSuccess(RegistrationDescription registrationDescription) throws Exception {
+                super.onSuccess(registrationDescription);
+                tryToSetUIInteraction(true);
+                registrationForm.setVisibility(View.VISIBLE);
+                loadingIndicator.setVisibility(View.GONE);
+                setupRegistrationForm(registrationDescription);
+            }
+
+            @Override
+            protected void onException(Exception ex) {
+                super.onException(ex);
+                tryToSetUIInteraction(true);
+                registrationForm.setVisibility(View.GONE);
+                loadingIndicator.setVisibility(View.GONE);
```

Comment:
3 lines above are somewhat same in the `onSuccess` callback, so, it'd be better to move them into a utility function.